include/RAJA/policy/sycl/MemUtils_SYCL.hpp

```diff
@@ -0,0 +1,80 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   Header file defining prototypes for routines used to manage
+ *          memory for SYCL reductions and other operations.
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_MemUtils_SYCL_HPP
+#define RAJA_MemUtils_SYCL_HPP
+
+#include "RAJA/config.hpp"
+
+#if defined(RAJA_ENABLE_SYCL)
+
+#include <CL/sycl.hpp>
+
+#include <cassert>
+#include <cstddef>
+#include <cstdio>
+#include <type_traits>
+#include <unordered_map>
+
+#include "RAJA/util/basic_mempool.hpp"
+#include "RAJA/util/mutex.hpp"
+#include "RAJA/util/types.hpp"
+
+#include "RAJA/policy/sycl/policy.hpp"
+
+namespace RAJA
+{
+
+namespace sycl
+{
+
+namespace detail
+{
+
+//! struct containing data necessary to coordinate kernel launches with reducers
+struct syclInfo {
+  sycl_dim_t gridDim{0};
+  sycl_dim_t blockDim{0};
+  cl::sycl::queue qu = cl::sycl::queue();
+  bool setup_reducers = false;
+#if defined(RAJA_ENABLE_OPENMP) && defined(_OPENMP)
+  syclInfo* thread_states = nullptr;
+  omp::mutex lock;
+#endif
+};
+
+extern syclInfo g_status;
+
+extern syclInfo tl_status;
```

Comment:
This declaration is incompatible with the definition when openmp is also enabled.

Reply:
WIll also include this in the issue.